packages/react/src/experimental/Await.tsx

```diff
@@ -0,0 +1,48 @@
+import { FunctionComponent, useEffect, useMemo } from 'react'
+import { useRerender } from '../hooks'
+import { Tuple } from '../types'
+import { suspensiveCache } from './suspensiveCache'
+
+type Awaited<TData = unknown> = { data: TData; reset: () => void }
+type AwaitOptions<TData, TKey extends Tuple> = {
+  key: TKey
+  fn: (options: { key: TKey }) => Promise<TData>
+}
+
+/**
+ * @experimental This is experimental feature.
+ */
+export const awaitOptions = <TData, TKey extends Tuple>(options: AwaitOptions<TData, TKey>) => options
+
+/**
+ * @experimental This is experimental feature.
+ */
+export const useAwait = <TData, TKey extends Tuple>(options: AwaitOptions<TData, TKey>): Awaited<TData> => {
+  const data = suspensiveCache.suspend(options.key, () => options.fn({ key: options.key }))
+
+  const rerender = useRerender()
+  const stringifiedKey = JSON.stringify(options.key)
+
+  useEffect(() => suspensiveCache.attach(options.key, rerender), [stringifiedKey, rerender])
+
+  return useMemo(
+    () => ({
+      data,
+      reset: () => suspensiveCache.reset(options.key),
+    }),
+    [stringifiedKey, data]
+  )
+}
+
+type AwaitProps<TData, TKey extends Tuple> = {
+  options: AwaitOptions<TData, TKey>
+  children: FunctionComponent<Awaited<TData>>
+}
+
+/**
+ * @experimental This is experimental feature.
+ */
+export const Await = <TData, TKey extends Tuple>({ children: Children, options }: AwaitProps<TData, TKey>) => {
```

Comment:
The function of `detach` is defined in the `suspensiveCache`, but I wonder if it is not necessary to use it in the Await component

Reply:
`detach` will be invoked on `useAwait`'s unmount internally